### PR TITLE
Use nbsphinx to execute demo notebook

### PR DESCRIPTION
**Pull request by maxrjones**
**Description of proposed changes**

Automatically execute the demo notebook when building the docs.

Fixes #

### CODE REVIEW

**Review by andersy005 — APPROVED** [review]
Nice, 👍🏽

File: doc/conf.py [hunk 1]
```diff
@@ -56,7 +56,7 @@
 
 # never execute notebooks: avoids lots of expensive imports on rtd
```
Comment (weiji14) [inline doc/conf.py:57]:
```suggestion
# only execute notebooks without any outputs
```

File: ci/requirements/doc.yml [hunk 2]
```diff
@@ -17,6 +15,12 @@ dependencies:
   - sphinx-autosummary-accessors
   - sphinx-copybutton
   - xarray
+  # For examples
+  - adlfs
+  - nbsphinx
+  - ipykernel
```
Comment (weiji14) [inline ci/requirements/doc.yml:21]: Alphabetical nitpick.
```suggestion
  - ipykernel
  - nbsphinx
```

**Review by weiji14 — APPROVED** [review]
Just some minor comments, otherwise good! For reference, readthedocs build at https://readthedocs.org/projects/xbatcher/builds/18654567/ took 4min 13sec.

**Comment by maxrjones** [comment]
> Just some minor comments, otherwise good! For reference, readthedocs build at https://readthedocs.org/projects/xbatcher/builds/18654567/ took 4min 13sec.

Thanks Wei Ji! It'll be interesting to see the speedup after fixing https://github.com/xarray-contrib/xbatcher/issues/121.

**Pull request merged**